pkg/sql/crdb_internal.go

```diff
@@ -1220,6 +1222,54 @@ CREATE TABLE crdb_internal.create_statements (
 								return err
 							}
 						}
+						// Create CREATE INDEX commands for INTERLEAVE tables. These commands
+						// are included in the ALTER TABLE statements.
+						if len(idx.Interleave.Ancestors) > 0 {
+							f := tree.NewFmtCtx(tree.FmtSimple)
+							intl := idx.Interleave
+							// Get the parent table information.
+							parentTableID := intl.Ancestors[len(intl.Ancestors)-1].TableID
+							var parentName tree.TableName
+							if lCtx != nil {
+								parentTable, err := lCtx.getTableByID(parentTableID)
+								if err != nil {
+									return err
+								}
+								parentDbDesc, err := lCtx.getDatabaseByID(parentTable.ParentID)
+								if err != nil {
+									return err
+								}
+								parentName = tree.MakeTableName(tree.Name(parentDbDesc.Name), tree.Name(parentTable.Name))
+								parentName.ExplicitSchema = parentDbDesc.Name != contextName
+							} else {
+								parentName = tree.MakeTableName(tree.Name(""), tree.Name(fmt.Sprintf("[%d as parent]", parentTableID)))
+								parentName.ExplicitCatalog = false
+								parentName.ExplicitSchema = false
+							}
+							var sharedPrefixLen int
+							for _, ancestor := range intl.Ancestors {
+								sharedPrefixLen += int(ancestor.SharedPrefixLen)
+							}
+							// Write the CREATE INDEX statements.
+							f.WriteString("CREATE ")
+							f.WriteString(idx.SQLString(&parentName))
+							f.WriteString(" INTERLEAVE IN PARENT ")
+							fmtCtx := tree.NewFmtCtx(tree.FmtSimple)
+							fmtCtx.FormatNode(&parentName)
+							f.WriteString(fmtCtx.CloseAndGetString())
+							f.WriteString(" (")
+							// Get all of the columns and write them.
+							for i, name := range idx.ColumnNames[:sharedPrefixLen] {
+								if i > 0 {
+									f.WriteString(", ")
```

Comment:
This doesn't seem to take into account column directions, which I assume interleaved indexes could be? If so, we need to add a test that has some ASC and DESC columns. Also use `idx.ColNamesFormat` which handles this.